lib/internal/url.js

```diff
@@ -932,6 +935,60 @@ function urlToOptions(url) {
   return options;
 }
 
+function getPathFromURLWin32(url) {
+  var hostname = url.hostname;
+  var pathname = url.pathname;
+  if (hostname !== '') {
+    // If hostname is set, then we have a UNC path
```

Comment:
Actually... nevermind about the pct escapes... for hostnames, that occurs while the URL is being parsed. There is a chance, however, that the hostname would be punycode encoded so I'll need to adjust for that... although I'm not entirely sure about the current support for extended characters in UNC hostnames.